spec/models/health/patient_referral_spec.rb

```diff
@@ -0,0 +1,60 @@
+require 'rails_helper'
+
+RSpec.describe Health::PatientReferral, type: :model do
+  describe 'derive referrals' do
+    it 'leaves simple open referrals alone' do
+      open_referral = create :patient_referral
+      expect(Health::PatientReferral.count).to eq(1)
+
+      Health::PatientReferral.build_derived_referrals(open_referral.compute_enrollment_changes).map(&:save!)
+      expect(Health::PatientReferral.count).to eq(1)
+    end
+
+    it 'leaves simple closed referrals alone' do
+      Timecop.freeze(Date.current - 91.days) do
+        closed_referral = create :patient_referral
+        Timecop.travel(Date.current + 91.days)
+        closed_referral.update(disenrollment_date: Date.current)
+        expect(closed_referral.versions.count).to eq(2)
+        expect(Health::PatientReferral.count).to eq(1)
+
+        Health::PatientReferral.build_derived_referrals(closed_referral.compute_enrollment_changes).map(&:save!)
+        expect(Health::PatientReferral.count).to eq(1)
+      end
+    end
+
+    it 'creates two contributing referrals  (one current) for re-opened referrals in 90 days' do
+      Timecop.freeze(Date.current - 91.days) do
```

Comment:
Timecop is nice here but unless I'm missing something (where maybe you are relying on the updated_at timestamp, I think you could just pass in a date with delta days to `disenrollment_date:`.  No harm in what you are doing, and I like the the entire world gets moved around, so oddities from that would be caught.

Reply:
Recovering previous enrollment dates from PaperTrail requires the `updated_at` timestamp, so using Timecop here allows PaperTrail to create an appropriate history.